File: stack_queue/leetcode20.java
import java.util.*;
public class leetcode20
{
    public static void main(String args[])
    {
        String s="{[(])}";
        boolean result=isValidParentheses(s);
        if(result)
            System.out.println("It is valid parentheses");
        else
            System.out.println("It is not valid parentheses");
    }

    public static boolean isValidParentheses(String s)
    {
        Stack <Character> st=new Stack<Character>();

        for(int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
            if(c=='{'||c=='['||c=='(')
                st.push(c);
            else
            {
                if(st.empty())
                    return false;
                else
                {
                    char peek=st.peek();
                    if((c=='}'&&peek=='{')||(c==')'&&peek=='(')||(c==']'&&peek=='['))
                        st.pop();
                    else
                        return false;
                }
            }
        }
        return st.empty();
    }
}

File: stack_queue/leetcode20.py
def isValidParentheses(s:str)->bool:
    stack =[]
    for c in s:
        if c in "{[(":
            stack.append(c)
        else:
            if not stack:
                return False
            else:
                peek=stack[-1]
                if (c=='}' and peek=='{') or \
                   (c==']' and peek=='[') or \
                   (c==')' and peek=='('):
                   stack.pop()
                else:
                    return False
    return not stack



s="{[()]}"
result=isValidParentheses(s)
if result:
    print("It is valid pareentheses")
else:
    print("It is not valid parentheses")